aviary/subsystems/propulsion/propulsion_mission.py

```diff
@@ -126,62 +135,89 @@ def setup(self):
-                + '_'
-                + str(i),
-            )
-            self.connect(
-                engine.name + '.' + Dynamic.Mission.NOX_RATE,
-                'vectorize_performance.' + Dynamic.Mission.NOX_RATE + '_' + str(i),
-            )
-
-            if getattr(engine, 'use_t4', False):
-                self.connect(
-                    engine.name + '.' + Dynamic.Mission.TEMPERATURE_T4,
-                    'vectorize_performance.'
-                    + Dynamic.Mission.TEMPERATURE_T4
-                    + '_'
-                    + str(i),
-                )
-
-            if getattr(engine, 'use_shp', False):
-                self.connect(
-                    engine.name + '.' + Dynamic.Mission.SHAFT_POWER,
-                    'vectorize_performance.'
-                    + Dynamic.Mission.SHAFT_POWER
-                    + '_'
-                    + str(i),
-                )
-                # self.connect(engine.name + '.' + Dynamic.Mission.SHAFT_POWER_CORRECTED,
-                #              'vectorize_performance.' + Dynamic.Mission.SHAFT_POWER_CORRECTED + '_' + str(i))
-
         self.add_subsystem(
             'propulsion_sum',
             subsys=PropulsionSum(num_nodes=nn, aviary_options=options),
             promotes_inputs=['*'],
             promotes_outputs=['*'],
         )
 
+    def configure(self):
+        # Special configure step needed to handle multiple, unique engine models.
+        # Handle checking each EngineModel for compatible outputs with
+        # vectorize_performance component and connecting those outputs
+
+        # TODO this list shouldn't be hardcoded so it can be extended by users
+        supported_outputs = [
+            Dynamic.Mission.ELECTRIC_POWER_IN,
+            Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE,
+            Dynamic.Mission.NOX_RATE,
+            Dynamic.Mission.SHAFT_POWER,
+            Dynamic.Mission.SHAFT_POWER_MAX,
+            Dynamic.Mission.TEMPERATURE_T4,
+            Dynamic.Mission.THRUST,
+            Dynamic.Mission.THRUST_MAX,
+        ]
+
+        engine_models = self.options['engine_models']
+        engine_names = [engine.name for engine in engine_models]
+        # num_engine_type = len(engine_models)
+
+        # determine if openMDAO messages and warnings should be suppressed
+        verbosity = self.options['aviary_options'].get_val(Settings.VERBOSITY)
+        out_stream = None
+        # DEBUG
+        if verbosity.value > 2:
+            out_stream = sys.stdout
+
+        comp_list = [self._get_subsystem(engine) for engine in engine_names]
+
+        # dictionaries of outputs for each engine in prop mission
+        output_dict = {}
+        # Dictionary of all unique inputs/outputs from all new components, keys are
+        # units for each var
+        unique_outputs = {}
+
+        # idx to be used for slicing inputs in next round of improvements
+        for idx, comp in enumerate(comp_list):
```

Comment:
instead of using a list, which will require us to keep track of the order we added the engines in, would it be better to use a dictionary with the name as the key and use `_2`, `_3`... for engine models with the same name?

Reply:
I'm not sure I follow, a dictionary won't solve any ordering problems (idx will be needed for `om.slicer`).
Order shouldn't be a problem because the list of engine models passed through options will never be changed (engine model list is saved as an attribute of the propulsion builder and never gets modified once initialized).